src/app/features/location/address-edit/address-edit.component.ts

```diff
@@ -74,6 +83,27 @@ export class AddressEditComponent {
     });
   }
 
+  async updateLocationFromGps() {
+    this.gpsLoading = true;
+    try {
+      const location = await this.gpsService.getGpsLocationCoordinates();
+      const address = await this.gpsService.getGpsLocationAddress();
```

Comment:
To ensure the exact same behavior, let's somehow reuse the logic for lookup between MapPopupComponent `mapClicked()` handler and this case here where the raw coordinates are coming from GPS (rather than the mouse click as in mapClicked).

I am not completely sure what's the best way to do this. Maybe moving the lookupCoordinates method from MapPopupComponent into the service?

We could also keep GpsService independent of any address lookup, just focusing on the coordinates

Reply:
I've updated the code to reuse the mapClicked() function for retrieving the address based on coordinates and to update the location accordingly. Could you please review the changes and let me know if any improvements are needed?